app/models/modal-window.ts

```diff
@@ -10,10 +10,16 @@
 export interface ModalInfo {
     readonly id: number;
     readonly selector: string;
+    readonly type: ModalType
     data: any;
 }
 
+export enum ModalType {
+    MODAL, // Will display as a drop down on macOS
+    CHILD, 
+} 
+
 export interface ModalService {
-    createModal(selector: string): [ModalInfo, Promise<any>];
+    createModal(selector: string, type: ModalType): [ModalInfo, Promise<any>];
```

Comment:
I've found that a Promise<any> allowed Typescript to make some bad inference that ended up messing with my compiles. I tried using specific types (even if it was a type that in effect was just `BadType` so that it forced a cast and I had to think about what I was doing).